dotnet/Vaas/src/Vaas/Authentication/TokenReceiver.cs

```diff
@@ -109,36 +109,36 @@ private async Task<TokenResponse> RequestTokenAsync(CancellationToken cancellati
 
     private FormUrlEncodedContent TokenRequestToForm()
```

Comment:
Consider making this class abstract, and just implement `TokenRequestToForm` in `ClientCredentialsGrantAuthenticator`/`ResourceOwnerPasswordGrantAuthenticator`, thus getting rid of the switch and logically separating the two.